src/workers/event.rs

```diff
@@ -853,6 +864,39 @@ where
     }
 }
 
+/// Helper that runs any iterable of futures and returns a list of commands and the proper result to
+/// use as a response (Ok if there were no errors, all of the errors combined otherwise)
+async fn get_commands_and_result<Fut, I>(futs: I, error_message: &str) -> (Vec<Command>, Result<()>)
+where
+    Fut: futures::Future<Output = Result<Vec<Command>>>,
+    I: IntoIterator<Item = Fut>,
+{
+    let (commands, errors): (Vec<_>, Vec<_>) = futures::future::join_all(futs)
+        .await
+        .into_iter()
+        .partition(Result::is_ok);
+    // SAFETY: We partitioned by Ok and Err, so the unwraps are fine
+    (
+        commands.into_iter().flat_map(Result::unwrap).collect(),
```

Comment:
I assume this is right, but having a hard time parsing why it's `flat_map` instead of `map`

Reply:
The inner type of each result is `Vec<Command>`, so this shortcuts it from `commands.into_iter().map(Result::unwrap).flatten().collect()`. I think it was clippy who told me not to do that